Source/Scene/TileBoundingS2Cell.js

```diff
@@ -0,0 +1,558 @@
+  var vertexCartographicScratch;
+  var distanceScratch;
+  var i;
+  for (i = 0; i < 4; i++) {
+    vertexScratch = s2Cell.getVertex(i);
+    vertexCartographicScratch = Cartographic.fromCartesian(vertexScratch);
+    vertexCartographicScratch.height = minimumHeight;
+    distanceScratch = Plane.getPointDistance(
+      topPlane,
+      Cartographic.toCartesian(vertexCartographicScratch)
+    );
+    if (Math.abs(distanceScratch) > maxDistance) {
+      maxDistance = distanceScratch;
+    }
+  }
+  var bottomPlane = Plane.clone(topPlane);
+  bottomPlane.normal = Cartesian3.negate(
+    bottomPlane.normal,
+    bottomPlane.normal
+  );
+  bottomPlane.distance -= maxDistance;
+  bottomPlane.distance = -bottomPlane.distance;
+
+  // Compute side planes.
+  // - Iterate through vertices
+  //   - Get a vertex and another vertex adjacent to it.
+  //   - Compute midpoint of geodesic between two vertices.
+  //   - Compute geodetic surface normal at center point.
+  //   - Compute vector between vertices.
+  //   - Compute normal of side plane. (cross product of top dir and side dir)
+  var adjacentVertexScratch = new Cartesian3();
+  var sideGeodesicScratch = new Cartesian3();
+  var sideMidpointScratch = new Cartesian3();
+  var topVectorScratch = new Cartesian3();
+  var sideVectorScratch = new Cartesian3();
+  var sideNormalScratch = new Cartesian3();
+  var sidePlanes = [];
+  for (i = 0; i < 4; i++) {
+    vertexScratch = s2Cell.getVertex(i);
+    adjacentVertexScratch = s2Cell.getVertex([i + 1] % 4);
+    sideGeodesicScratch = new EllipsoidGeodesic(
+      Cartographic.fromCartesian(vertexScratch),
+      Cartographic.fromCartesian(adjacentVertexScratch)
+    );
+    sideMidpointScratch = Cartographic.toCartesian(
+      sideGeodesicScratch.interpolateUsingFraction(0.5)
+    );
+    topVectorScratch = Ellipsoid.WGS84.geodeticSurfaceNormal(
+      sideMidpointScratch
+    );
+    Cartesian3.normalize(
+      Cartesian3.subtract(
+        adjacentVertexScratch,
+        vertexScratch,
+        new Cartesian3()
+      ),
+      sideVectorScratch
+    );
+    Cartesian3.normalize(
+      Cartesian3.cross(topVectorScratch, sideVectorScratch, new Cartesian3()),
+      sideNormalScratch
+    );
+
+    sidePlanes[i] = Plane.fromPointNormal(
+      sideMidpointScratch,
+      sideNormalScratch
+    );
+  }
+
+  return [topPlane, bottomPlane, sidePlanes];
+}
+
+var n0 = new Cartesian3();
+var n1 = new Cartesian3();
+var n2 = new Cartesian3();
+var x0 = new Cartesian3();
+var x1 = new Cartesian3();
+var x2 = new Cartesian3();
+var t0 = new Cartesian3();
+var t1 = new Cartesian3();
+var t2 = new Cartesian3();
+var f0 = new Cartesian3();
+var f1 = new Cartesian3();
+var f2 = new Cartesian3();
+var s = new Cartesian3();
+/**
+ * Computes intersection of 3 planes.
+ * @private
+ */
+function computeIntersection(p0, p1, p2) {
+  n0 = p0.normal;
+  n1 = p1.normal;
+  n2 = p2.normal;
+
+  Cartesian3.multiplyByScalar(p0.normal, -p0.distance, x0);
```

Comment:
Again, this would be more idiomatic. Check this throughout this file.
```suggestion
  var x0 = Cartesian3.multiplyByScalar(p0.normal, -p0.distance, scratchX0);
```

See also https://github.com/CesiumGS/cesium/blob/master/Documentation/Contributors/CodingGuide/README.md#result-parameters-and-scratch-variables